docker/service.go

```diff
@@ -46,15 +59,20 @@ func (s *Service) collectContainers() ([]*Container, error) {
 
 	result := []*Container{}
 
+	if len(containers) == 0 {
```

Comment:
I'm not sure if it's an optimization or not. If `len(containers)` is equal to `0`, the `for` loop won't do a thing anyway right ? :stuck_out_tongue: